src/rpc/blockchain.cpp

```diff
@@ -2126,6 +2128,150 @@ static UniValue getcolor(const JSONRPCRequest& request)
+        }
+    };
+    while (pcursor->Valid()) {
+        if (iter % 5000 == 0) ShutdownRequested();
+        ++iter;
+        if (pcursor->GetKey(key) && pcursor->GetValue(coin)) {
+            if (key.hashMalFix != last_hash) {
+                write_coins_to_file(afile, last_hash, coins, written_coins_count);
+                last_hash = key.hashMalFix;
+                coins.clear();
+            }
+            coins.emplace_back(key.n, coin);
+        }
+        pcursor->Next();
+    }
+
+    if (!coins.empty()) {
+        write_coins_to_file(afile, last_hash, coins, written_coins_count);
+    }
+
+    afile.fclose();
+
+    return maybe_stats;
+}
+
+/**
+ * Serialize the UTXO set to a file for loading elsewhere.
+ *
+ * @see SnapshotMetadata
+ */
+static UniValue utxosnapshot(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+        "utxosnapshot \"path\""
+        "Write the serialized UTXO set to disk."
+        "\nArguments:\n"
+            "1. path     -   Path to the output file. If relative, will be prefixed by datadir.\n"
+         "\nResult:\n"
+                "coins_written  - the number of coins written in the snapshot\n"
+                "base_hash  -  the hash of the base of the snapshot\n"
+                "base_height  - the height of the base of the snapshot\n"
+                "path  - the absolute path that the snapshot was written to\n"
+                "txoutset_hash  - the hash of the UTXO set contents\n"
+                "nchaintx  - the number of transactions in the chain up to and including the base block\n");
+
+    std::string path_name = request.params[0].get_str();
+    const fs::path path = GetDataDir()/ path_name;
+    // Write to a temporary path and then move into `path` on completion
+    // to avoid confusion due to an interruption.
+    const fs::path temppath = GetDataDir() / path_name.append(".incomplete");
+
+    if (fs::exists(path)) {
+        throw JSONRPCError(
+            RPC_INVALID_PARAMETER,
+            "path already exists. If you are sure this is what you want, move it out of the way first");
+    }
+
+    FILE* file{fsbridge::fopen(temppath, "wb")};
+    CAutoFile afile{file, SER_DISK, CLIENT_VERSION};
+    if (afile.IsNull()) {
+        throw JSONRPCError(
+            RPC_INVALID_PARAMETER,
+            "Couldn't open file temp file for writing.");
+    }
+
+    CCoinsStats maybe_stats = CreateUTXOSnapshot(afile, path, temppath);
+    fs::rename(temppath, path);
+    const CBlockIndex* tip = LookupBlockIndex(maybe_stats.hashBlock);
+
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("coins_written", maybe_stats.nTransactionOutputs);
+    result.pushKV("base_hash", tip->GetBlockHash().ToString());
+    result.pushKV("base_height", tip->nHeight);
+    result.pushKV("path", path.c_str());
+    result.pushKV("txoutset_hash", maybe_stats.hashSerialized.ToString());
+    result.pushKV("nchaintx", tip->nChainTx);
```

Comment:
The `versionO and `network_id` mentioned in the [issue](https://github.com/chaintope/tapyrus-core/issues/290) are missing.

Reply:
These are part of the shapshot metadata written to the snapshot file. It is not in the RPC output. Test case `test/functional/rpc_dumptxoutset.py` verifies that these are written to the file by the RPC. IT then reads them from the file, deserialises and verifies that they match the expected output.